myia/impl_py.py

```diff
@@ -0,0 +1,160 @@
+"""Implementations for the debug VM."""
+
+
+from typing import Dict, Callable
+from copy import copy
+from myia import primops
+
+
+implementations: Dict[primops.Primitive, Callable] = {}
+
+
+def registrar(pfx, store):
+    """Create a decorator to associate a function to a primitive.
+
+    The function name will need to start with `pfx`, followed by
+    the primitive's name in `primops`. It will then be associated
+    to that Primitive in `store`.
+    """
+    def deco(fn):
+        """Decorate the function."""
+        name = fn.__name__
+        assert name.startswith(pfx)
+        name = name.split(pfx)[1]
+        prim = getattr(primops, name)
+        store[prim] = fn
+        return fn
+    return deco
+
+
+impl = registrar('impl_', implementations)
```

Comment:
`impl` isn't really a universal abbreviation, so maybe just write out `implementation`

Reply:
It's a pretty common abbreviation, I think. What else would it mean? Implication?

Reply:
Implicit, for example.